### DISCUSSION TITLE
Additional Support for Navigation in Viewport (Middle Mouse Button)

### DISCUSSION THREAD

**Discussion by RavenStryker** [OpenRCT2/OpenRCT2#22959] [Feature requests]
Is there a setting or add-on to change the navigation of the viewport in oRCT2? Like instead of right clicking to move around, can you use middle-mouse button and instead of it pushing the pulling camera in the direction of the mouse, can it push it instead like more modern 3d apps do (blender, parkitect, ect).

I don't mind modifying some code or changing somethings internally of the game ot get this functionality. Or if it can be made into a plugin that would be most ideal and I can learn how to do that if pointed in the right direction on how to do so. Thanks!

**Comment by janisozaur** [comment]
Would a simple 'invert mouse direction's for map movement be enough?

**Comment by RavenStryker** [comment]
That would be a nice start. Ideally being able to map a hotkey in this menu would be the best addition. I know it says "keyboard shortcuts", but it could still house navigation interaction keys. I know if you swap your primary mouse button from left to right in windows, the game identifies this and you instead use your left click to navigate around instead of your right click, so I would "assume" it could pick up middle mouse button as well, but maybe it is tied to a formula that solely works off of windows primary functionality. Adding an inverse mouse direction is a great start though and I would highly accept even just that. :)

![image](https://github.com/OpenRCT2/OpenRCT2/assets/20632191/0fbcce82-ebd9-4cd3-af81-966b19445b97)

**Comment by RavenStryker** [comment]
> Would a simple 'invert mouse direction's for map movement be enough?

Is the "invert mouse direction" something I can do myself or will it need to be patched in?

**Comment by 733737** [comment]
it is already in the config.ini file for you to change
